TemplePlus/python/python_object.cpp

```diff
@@ -2854,11 +2854,32 @@ static PyObject* PyObjHandle_GetIdxInt(PyObject* obj, PyObject* args) {
 	if (!PyArg_ParseTuple(args, "i|i:objhndl.obj_get_idx_int", &field, &subIdx)) {
 		return 0;
 	}
-	assert(subIdx >= 0);
-	auto value = objects.getArrayFieldInt32(self->handle, field, subIdx);
+	assert(subIdx >= -1); // return size of array if -1
```

Comment:
Please make a separate get_idx_int_size function. This looks like it could lead to nasty bugs.